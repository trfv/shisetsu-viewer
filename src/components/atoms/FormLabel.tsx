import Box from "@material-ui/core/Box";
import React, { FC } from "react";

type FormLabelProps = {
  labelText: string;
  fontSize?: string;
};

const FormLabel: FC<FormLabelProps> = ({ fontSize, labelText }: FormLabelProps) => {
  return <Box fontSize={fontSize}>{labelText}</Box>;
};

FormLabel.defaultProps = {
  fontSize: "0.8rem",
};

export default FormLabel;
