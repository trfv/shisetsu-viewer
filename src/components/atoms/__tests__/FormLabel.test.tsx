import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";
import FormLabel from "../FormLabel";

it("snapshot", () => {
  const props = {
    labelText: "formLabel",
    fontSize: "8px",
  };
  const component = renderer.create(<FormLabel {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("interaction", () => {
  const props = {
    labelText: "formLabel",
    fontSize: "8px",
  };
  const formLabel = render(<FormLabel {...props} />);
  expect(formLabel.queryByText("formLabel")).toBeTruthy();
  expect(formLabel.queryByText("formLabel")).toHaveStyle("font-size: 8px");
});
