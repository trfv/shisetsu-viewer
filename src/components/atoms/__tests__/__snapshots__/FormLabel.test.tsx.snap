// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-1"
>
  formLabel
</div>
`;
